File: searching/linear_search/cpp/linear_search.cpp
// Linear Search in C++
#include <iostream>
using namespace std;

int main() {

     // Random data to search from
     int searchArray[] = {10, 23, 89, 126, 51, 7, 45, 55};
     int size = sizeof(searchArray) / sizeof(searchArray[0]);
     int search = 51;    // Number to search
     int index = -1;

     for (int i = 0; i < size; i++) {
          if (searchArray[i] == search) {
               index = i;
               break;
          }
     }

     if (index >= 0)
          cout << search << " found at index " << index << endl;
     else
          cout << search << " not found in the array" << endl;

     return 0;
}

File: searching/linear_search/rust/linear_search.rs
fn search(arr:[i32;5],key:i32) -> bool {
   for a in arr.iter() {
        if *a == key {
            return true;
        }
    }
    return false;
}

fn main() {

    // input_array
    let a: [i32; 5] = [1, 2, 3, 4, 5];
    
    println!("Found 5 in a: {}", search(a,5));
    println!("Found -5 in a: {}", search(a,-5));
    
}
